src/StreamJsonRpc/MessageFormatterHelper.cs

```diff
@@ -0,0 +1,24 @@
+﻿
+namespace StreamJsonRpc
+{
+    using System;
+    using System.Linq;
+    using System.Reflection;
+
+    public class MessageFormatterHelper
+    {
+        public static Type FindIProgressOfT(Type objectType)
+        {
+            if (objectType.IsConstructedGenericType && objectType.GetGenericTypeDefinition().Equals(typeof(IProgress<>)))
```

Comment:
>objectType [](start = 16, length = 10)

validate the `objectType` argument isn't null before dereferencing it.

Reply:
You're still dereferencing outside your if check. It's an error if the caller passes in null, so you can just write `Requires.NotNull(objectType, nameof(objectType));` at the top of the method.

---
In reply to: [302227109](https://github.com/milopezc/vs-streamjsonrpc/pull/1#discussion_r302227109) [](ancestors = 302227109)